src/transformers/models/distilbert/modeling_distilbert.py

```diff
@@ -322,11 +322,13 @@ def reshape(x: torch.Tensor) -> torch.Tensor:
         # in fp32. (LlamaRMSNorm handles it correctly)
 
         if query_states.dtype == torch.float32:
+            if torch.is_autocast_enabled():
+                target_dtype = torch.get_autocast_gpu_dtype()
             # Handle the case where the model is quantized
-            if hasattr(self.config, "_pre_quantization_dtype"):
+            elif hasattr(self.config, "_pre_quantization_dtype"):
                 target_dtype = self.config._pre_quantization_dtype
             else:
-                target_dtype = self.q_lin.weight.dtype
+                target_dtype = self.q_proj.weight.dtype
```

Comment:
q_proj does not exist
```suggestion
                target_dtype = self. q_lin.weight.dtype
```